### PR TITLE
Move all code related to zpages to the zpages extension

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This PR moves all the code related to zpages logic into the zpages extension itself.
There is no zpage code in the service package.
The zpage code still relies on casting host to get a struct representing the current structure of the pipelines, represented as simple types.
All other reporting logic of the zpagesextension is rendered internally to the extension using public APIs of `Host`.

### CODE REVIEW

File: component/graph.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package component // import "go.opentelemetry.io/collector/component"
+
+// Graph is a struct representing the pipelines
+// currently registered with the `Host`.
+type Graph struct {
```
Comment (bogdandrutu) [inline component/graph.go:8]: Please don't expose any new public API about our internal graph/pipelines.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
See the only main comment, which is we don't want to expose more public APIs about our internal representation.

File: component/graph.go (hunk 1 above)
Comment (atoulme) [reply component/graph.go:8]: OK, I'll expose it internally for now.

**Comment by atoulme** [comment]
> See the only main comment, which is we don't want to expose more public APIs about our internal representation.

I have removed any new APIs showing our internal representation.

**Pull request closed without merging**